src/analysis/special_functions.rs

```diff
@@ -44,6 +46,11 @@ fn update_func(func: &mut FuncInfo, type_: Type) -> bool {
     }
     // I assume `to_string` functions never return `NULL`
     if type_ == Type::ToString {
+        // Hack to make sure the names never, ever clash.
+        // TODO: Should perhaps merge ToStr and ToString, then store in
+        // specials whether this implementation returns pointers to 'static
+        // strings (which is possible but not guaranteed with `transfer none`)
```

Comment:
That sounds like a good idea. That way we could also store the actual function name (`get_name()` / `to_string()` and whatever else we might find at later time).

Reply:
It would beak `impl FromStr for Type` though, but then again we should probably insert such info in the `specials` `Map`.

Would make most of the code more generic though, which is my main goal here. Let's see what I can come up with.

Reply:
Implemented like that. Perhaps we should rename `ToString` to `Display` so that it's clear this is about the trait, that can be implemented by either `to_string` (renamed to `to_str` to prevent `ToString::to_string` clashes) or `get_name`.

Reply:
Yes naming it `Display` would make a lot more sense. Feel free to rename that